lib/TH/generic/THTensorMath.h

```diff
@@ -128,6 +128,11 @@ TH_API void THTensor_(eqTensorT)(THTensor *r_, THTensor *ta, THTensor *tb);
 TH_API void THTensor_(abs)(THTensor *r_, THTensor *t);
 #endif
 
+#if defined(TH_REAL_IS_FLOAT) || defined(TH_REAL_IS_DOUBLE) || defined(TH_REAL_IS_HALF)
```

Comment:
Why the change here?  If I'm reading this right, even if GENERIC_USE_HALF_MATH were enabled, we wouldn't generate these because in THTensorMath.c, these are still guarded by TH_REAL_IS_FLOAT/DOUBLE and not HALF.  Did you mean to change THTensorMath.c as well?